targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F412xG/TARGET_MTB_ADV_WISE_1530/PinNames.h

```diff
@@ -0,0 +1,241 @@
+    PD_9  = 0x39,
+    PD_10 = 0x3A,
+    PD_11 = 0x3B,
+    PD_12 = 0x3C,
+    PD_13 = 0x3D,
+    PD_14 = 0x3E,
+    PD_15 = 0x3F,
+
+    PE_0  = 0x40,
+    PE_1  = 0x41,
+    PE_2  = 0x42,
+    PE_3  = 0x43,
+    PE_4  = 0x44,
+    PE_5  = 0x45,
+    PE_6  = 0x46,
+    PE_7  = 0x47,
+    PE_8  = 0x48,
+    PE_9  = 0x49,
+    PE_10 = 0x4A,
+    PE_11 = 0x4B,
+    PE_12 = 0x4C,
+    PE_13 = 0x4D,
+    PE_14 = 0x4E,
+    PE_15 = 0x4F,
+
+    PF_0  = 0x50,
+    PF_1  = 0x51,
+    PF_2  = 0x52,
+    PF_3  = 0x53,
+    PF_4  = 0x54,
+    PF_5  = 0x55,
+    PF_6  = 0x56,
+    PF_7  = 0x57,
+    PF_8  = 0x58,
+    PF_9  = 0x59,
+    PF_10 = 0x5A,
+    PF_11 = 0x5B,
+    PF_12 = 0x5C,
+    PF_13 = 0x5D,
+    PF_14 = 0x5E,
+    PF_15 = 0x5F,
+
+    PG_0  = 0x60,
+    PG_1  = 0x61,
+    PG_2  = 0x62,
+    PG_3  = 0x63,
+    PG_4  = 0x64,
+    PG_5  = 0x65,
+    PG_6  = 0x66,
+    PG_7  = 0x67,
+    PG_8  = 0x68,
+    PG_9  = 0x69,
+    PG_10 = 0x6A,
+    PG_11 = 0x6B,
+    PG_12 = 0x6C,
+    PG_13 = 0x6D,
+    PG_14 = 0x6E,
+    PG_15 = 0x6F,
+
+    PH_0  = 0x70,
+    PH_1  = 0x71,
+
+    // ADC internal channels
+    ADC_TEMP = 0xF0,
+    ADC_VREF = 0xF1,
+    ADC_VBAT = 0xF2,
+
+    // Arduino connector namings
+    A0          = PA_3,
+    A1          = PC_0,
+    A2          = PC_3,
+    A3          = PC_1,
+    A4          = PC_4,
+    A5          = PC_5,
+    D0          = PG_9,
+    D1          = PG_14,
+    D2          = PF_15,
+    D3          = PE_13,
+    D4          = PF_14,
+    D5          = PE_11,
+    D6          = PE_9,
+    D7          = PF_13,
+    D8          = PF_12,
+    D9          = PD_15,
+    D10         = PD_14,
+    D11         = PA_7,
+    D12         = PA_6,
+    D13         = PA_5,
+    D14         = PB_9,
+    D15         = PB_8,
+
+    // Generic signals namings
+    GPIO0       = PA_5,
+    GPIO1       = PB_0,
+    GPIO2       = PB_1,
+    GPIO7       = PB_14,
+
+    I2S_CK		= PB_12,
+    I2S_SW		= PA_4,
+    I2S_MCK		= PB_10,
+    I2S_SD		= PC_12,
+
+    SERIAL_TX   = PA_9,
+    SERIAL_RX   = PA_10,
+
+    UART2_TX    = PB_10,
+    UART2_RX    = PC_11,
+#ifdef MBED_CONF_TARGET_STDIO_UART_TX
+    STDIO_UART_TX = MBED_CONF_TARGET_STDIO_UART_TX,
+#else
+    STDIO_UART_TX = SERIAL_TX,
+#endif
+
+#ifdef MBED_CONF_TARGET_STDIO_UART_RX
+    STDIO_UART_RX = MBED_CONF_TARGET_STDIO_UART_RX,
+#else
+    STDIO_UART_RX = SERIAL_RX,
+#endif
+
+    USBRX      = STDIO_UART_RX,
+    USBTX      = STDIO_UART_TX,
+
+    I2C0_SCL    = PB_6,
+    I2C0_SDA    = PB_7,
+    I2C0_RST    = PC_13,
+
+    SPI0_MOSI   = PC_3,
+    SPI0_MISO   = PC_2,
+    SPI0_SCK    = PB_13,
+    SPI0_CS     = PB_9,
+
+    PWM_OUT     = D9,
+
+    ADCONV0     = PA_6,
+    ADCONV1     = PA_7,
+    ADCONV2     = PC_4,
+    ADCONV3     = PC_5,
+
+    EXTI1       = PC_0,
+
+    USB_HS_DP   = PA_12,
+    USB_HS_DN   = PA_11,
+    LED_RED     = GPIO0,
+    LED1 = LED_RED,
+    LED2 = LED_RED,
+    USER_BUTTON = GPIO2,
+    // Not connected
+    NC = (int)0xFFFFFFFF
```

Comment:
All pins from the module must be defined. Even the non-connected ones. You may want to move the NC to the top for this.